Add "Quick Settings Switchs for Plugins"

Hugo-COLLIN opened the pull request:
# I am submitting a new Community Plugin : <br>Quick Settings Switchs for Plugins

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[import * as fs from "fs";](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/fb422de6bae00414e6ef1a64aca27c0147f3b3c5/src/main.ts#L2-L3
These imports are only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[plugin.style.display = 'flex';](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/fb422de6bae00414e6ef1a64aca27c0147f3b3c5/src/main.ts#L89-L91)
You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.

[async savePluginState(app: App, pluginId: string, enabled: boolean) {](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/fb422de6bae00414e6ef1a64aca27c0147f3b3c5/src/main.ts#L133-L156)
There are additional `enablePluginAndSave`/`disablePluginAndSave` functions that you can use to replace this.

[containerEl.createEl("h2", {text: "Plugin Toggle Settings"});](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/fb422de6bae00414e6ef1a64aca27c0147f3b3c5/src/settingsTab.ts#L21)
[Don't include a header with the plugin name in the settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

What is your settings tab for?
It looks like it's the same content as the Community plugins tab.

Hugo-COLLIN commented:
@joethei Thank you for your review.

As you suggested, I used `enablePluginAndSave` and `disablePluginAndSave`, so I don't need to import fs anymore and I can keep `isDesktopOnly` marked as `false`.

I also moved my styles in the styles.css file.

The settings tab was just for testing, I didn't import this part of the code in main.ts. Now I deleted it for more convenience.

Hugo-COLLIN commented:
Don't understand why checks are failing.

joethei commented:
That's a git error we see showing up every now and again, not sure what is causing it.

Hugo-COLLIN commented:
Hello @joethei @liamcain,
A simple message to ask you if it's normal the revision lasts so long : it's about 1 month now!
I understand you have a lot of plugins to check. My only question is to kwow when my plugin will be reviewed.
I know validating plugins is complex and time-consuming, so thank you for your work!

liamcain commented:
- [registerDomEvent](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/39d1af035363aff9fe88dbba2791094021a35adb/src/main.ts#L13) why are you reregistering this event every time the layout changes? This will keep adding more and more event listeners to the sidedock. You are also not checking to make sure that the querySelector resolves. If this returns null, then the registerDomEvent will error
- [innerHTML](https://github.com/Hugo-COLLIN/obsidian-quick-settings-switchs/blob/39d1af035363aff9fe88dbba2791094021a35adb/src/main.ts#L22) You do not need to inject your css this way. Just add the CSS to styles.css
